<launch>
    <!-- ========================================= -->
    <arg name="use_sim_time" default="true"/>
    <arg name="log_level" default="warn"/>
    <arg name="config_path" default="$(find-pkg-share sign_detector)/config/config.yaml"/>

    <!-- ========================================= -->
    <node name="eagleye_pose" pkg="sign_detector" exec="fix_to_pose_node" output="log" args="--ros-args --log-level $(var log_level)">
        <param name="fix_topic" value="/eagleye/fix"/>
        <param name="pose_topic" value="/eagleye/pose"/>
        <param name="map_topic" value="/map/vector_map"/>
        <param name="use_sim_time" value="$(var use_sim_time)"/>
    </node>

    <node name="raw_pose" pkg="sign_detector" exec="fix_to_pose_node" output="log" args="--ros-args --log-level $(var log_level)">
        <param name="fix_topic" value="/sensing/gnss/ublox/nav_sat_fix"/>
        <param name="pose_topic" value="/sensing/gnss/ublox/pose"/>
        <param name="map_topic" value="/map/vector_map"/>
        <param name="use_sim_time" value="$(var use_sim_time)"/>
    </node>

    <node name="pose_to_path" pkg="sign_detector" exec="pose_to_path_node" output="log" args="--ros-args --log-level $(var log_level)">
        <param name="sub_topics" value="[/sensing/gnss/ublox/pose, /localization/pose_estimator/pose, /eagleye/pose]"/>
        <param name="pub_topics" value="[/sensing/gnss/ublox/path, /localization/pose_estimator/path, /eagleye/path]"/>
        <param name="use_sim_time" value="$(var use_sim_time)"/>
    </node>

    <node name="ll2_to_image" pkg="sign_detector" exec="ll2_to_image_node" output="log" args="--ros-args --log-level $(var log_level)">
        <param name="pose_topic" value="/eagleye/pose"/>
        <param name="map_topic" value="/map/vector_map"/>
        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <param name="max_range" value="20"/>
        <param name="image_size" value="800"/>
        <param name="line_thick" value="2"/>
    </node>

    <node name="lsd" pkg="sign_detector" exec="lsd_node" output="screen" args="--ros-args --log-level $(var log_level)">
        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <param name="config_path" value="$(var config_path)"/>
        <param name="line_thick" value="2"/>
    </node>
    <node name="imu_kf" pkg="sign_detector" exec="imu_kf_node" output="log" args="--ros-args --log-level $(var log_level)">
        <param name="use_sim_time" value="$(var use_sim_time)"/>
    </node>
    <node name="particle_filter" pkg="sign_detector" exec="particle_filter_node" output="screen" args="--ros-args --log-level $(var log_level)">
        <param name="use_sim_time" value="$(var use_sim_time)"/>
        <param name="blur_size" value="13"/>
        <param name="dilate_size" value="2"/>
    </node>
</launch>
